recipes-extended/xen/files/allow-mwait-cstate.patch

```diff
@@ -1,98 +0,0 @@
-################################################################################
```

Comment:
So I agree PV kernels can't actually execute MWAIT or MONITOR directly, I agree.  But doesn't this patch just advertise to PV guests whether the host supports it?  Not that it is claiming PV guests can execute the instruction?

I have to imagine at some point, someone wanted to be able to make a decision about deep C-states from inside a PV guest, perhaps by querying the cpuid when otherwise such a query would not say whether the host supported MONITOR/MWAIT.  Whether or not we have anything making use of such a query is up for debate.  I'd have to grep and get lucky I think.

Reply:
I agree that at some point, some PV kernel must have used this.  But we don't have a patch in our tree to do something based on it.  Any other distro is upstream-based these days, and per the comment shouldn't be using this.

Dom0 is still allowed, and that is the one that can do some C-state uploading.

So it just seems like something we should drop.  If only someone documented the *why* for this patch originally...

Reply:
Agreed, missing a lot of context from the original commit pre-Github.  We can do regression testing and keep an eye on this, but yea for the most part I doubt anything is using this capability.

Reply:
I'll expand the commit message with: "At some point, some PV kernel must have used this, but we don't have a patch in our tree to do something based on it. Any other distro is upstream-based these days, and per the comment shouldn't be using this.  Dom0 is still allowed, and that is the one that can do some C-state uploading."